java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java

```diff
@@ -0,0 +1,142 @@
+    private JavassistClassDeclaration classDeclarationSubClassIncludedJar;
+    private JavassistClassDeclaration classDeclarationInterfaceUserIncludedJar;
+    private JavassistClassDeclaration classDeclarationSubClassExcludedJar;
+    private JavassistClassDeclaration classDeclarationInterfaceUserExcludedJar;
+
+    @Before
+    public void setup() throws IOException {
+        final String pathToMainJar = adaptPath("src/test/resources/javassist_symbols/main_jar/main_jar.jar");
+        final String pathToIncludedJar = adaptPath("src/test/resources/javassist_symbols/included_jar/included_jar.jar");
+        typeSolver = new CombinedTypeSolver(new JarTypeSolver(pathToIncludedJar), new JarTypeSolver(pathToMainJar), new ReflectionTypeSolver());
+
+        symbolSolver = new SymbolSolver(typeSolver);
+
+        classDeclarationConcreteClass = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.ConcreteClass");
+        classDeclarationSubClassOwnJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.SubClassOwnJar");
+        classDeclarationSubClassIncludedJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.SubClassIncludedJar");
+        classDeclarationSubClassExcludedJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.SubClassExcludedJar");
+        classDeclarationInterfaceUserOwnJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.InterfaceUserOwnJar");
+        classDeclarationInterfaceUserIncludedJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.InterfaceUserIncludedJar");
+        classDeclarationInterfaceUserExcludedJar = (JavassistClassDeclaration) typeSolver.solveType("com.github.javaparser.javasymbolsolver.javassist_symbols.main_jar.InterfaceUserExcludedJar");
+    }
+
+    @Test
+    public void testSolveSymbolInTypeCanSolveFirstOwnField() {
+        assertCanSolveSymbol("STATIC_STRING", classDeclarationConcreteClass);
+    }
+
+    @Test
+    public void testSolveSymbolInTypeCanSolveSecondOwnField() {
+        assertCanSolveSymbol("SECOND_STRING", classDeclarationConcreteClass);
+    }
+
+    @Test
+    public void testSolveSymbolInTypeCantResolveNonExistantField() {
+        SymbolReference<? extends ResolvedValueDeclaration> solvedSymbol = symbolSolver.solveSymbolInType(classDeclarationConcreteClass, "FIELD_THAT_DOES_NOT_EXIST");
+
+        assertFalse(solvedSymbol.isSolved());
+
+        try {
+            solvedSymbol.getCorrespondingDeclaration();
+        } catch (Exception e) {
+            assertTrue(e instanceof UnsupportedOperationException);
+            assertNull(e.getMessage());
```

Comment:
Not sure if the message should be null also in the future

Reply:
How does `CorrespondingDeclaration not available for unsolved symbol` sound?

Reply:
It sounds good